hack/deploy-karmada.sh

```diff
@@ -116,19 +115,28 @@ function generate_cert_secret {
     sed -i'' -e "s/{{etcd_client_crt}}/${ETCD_CLIENT_CRT}/g" "${TEMP_PATH}"/karmada-cert-secret-tmp.yaml
     sed -i'' -e "s/{{etcd_client_key}}/${ETCD_CLIENT_KEY}/g" "${TEMP_PATH}"/karmada-cert-secret-tmp.yaml
 
-    sed -i'' -e "s/{{ca_crt}}/${karmada_ca}/g" "${TEMP_PATH}"/secret-tmp.yaml
-    sed -i'' -e "s/{{client_crt}}/${KARMADA_CRT}/g" "${TEMP_PATH}"/secret-tmp.yaml
-    sed -i'' -e "s/{{client_key}}/${KARMADA_KEY}/g" "${TEMP_PATH}"/secret-tmp.yaml
-
     sed -i'' -e "s/{{server_key}}/${KARMADA_KEY}/g" "${TEMP_PATH}"/karmada-webhook-cert-secret-tmp.yaml
     sed -i'' -e "s/{{server_certificate}}/${KARMADA_CRT}/g" "${TEMP_PATH}"/karmada-webhook-cert-secret-tmp.yaml
 
     kubectl --context="${HOST_CLUSTER_NAME}" apply -f "${TEMP_PATH}"/karmada-cert-secret-tmp.yaml
-    kubectl --context="${HOST_CLUSTER_NAME}" apply -f "${TEMP_PATH}"/secret-tmp.yaml
     kubectl --context="${HOST_CLUSTER_NAME}" apply -f "${TEMP_PATH}"/karmada-webhook-cert-secret-tmp.yaml
+
+    components=(aggregated-apiserver controller-manager kube-controller-manager scheduler descheduler metrics-adapter search webhook interpreter-webhook-example)
+    for component in "${components[@]}"
+    do
+      generate_config_secret ${component} ${karmada_ca} ${KARMADA_CRT} ${KARMADA_KEY}
+    done
+
     rm -rf "${TEMP_PATH}"
 }
 
+function generate_config_secret() {
+  export component=$1 ca_crt=$2 client_crt=$3 client_key=$4
+  envsubst < "${REPO_ROOT}"/artifacts/deploy/karmada-secret-config.yaml > "${TEMP_PATH}"/karmada-secret-config-${component}.yaml
```

Comment:
Seems you forgot to update the file path here.
The CI might fail because of this.